azure-pipelines/compliance.yml

```diff
@@ -9,32 +13,70 @@ pr:
       - dev/*
 
 pool:
-  vmImage: windows-latest
-
-steps:
-- task: PoliCheck@2
-  inputs:
-    targetType: 'F'
-    targetArgument: '$(Build.SourcesDirectory)'
-
-- task: ComponentGovernanceComponentDetection@0
-  displayName: 'Component Detection'
-
-- task: AntiMalware@4
-  inputs:
-    InputType: 'Basic'
-    ScanType: 'CustomScan'
-    FileDirPath: '$(Build.StagingDirectory)'
-    TreatSignatureUpdateFailureAs: 'Warning'
-    SignatureFreshness: 'UpToDate'
-    TreatStaleSignatureAs: 'Error'
-
-- task: CredScan@3
-  inputs:
-    scanFolder: '$(Build.SourcesDirectory)'
+  name: VSEngSS-MicroBuild2022-1ES
 
-- task: TSAUpload@2
-  displayName: 'TSA upload'
-  inputs:
-    GdnPublishTsaOnboard: True
-    GdnPublishTsaConfigFile: '$(Build.SourcesDirectory)/azure-pipelines/TSAOptions.json'
+jobs:
+- job: 'UnrealEngine_VisualStudioTools_Compliance'
+  timeoutInMinutes: 1440
+  steps:
+    - task: CmdLine@2
+      displayName: 'Clone Unreal Engine Repository'
+      inputs:
+        script: 'git clone "https://$(token)@github.com/EpicGames/UnrealEngine" --single-branch --branch $(ue_branch) --depth 1 ue'
+        workingDirectory: '$(Agent.BuildDirectory)'
+    - task: PowerShell@2
+      displayName: '[UE] Append /unattended option'
+      inputs:
+        targetType: 'inline'
+        script:
+          $filePath = "$(Agent.BuildDirectory)/ue/Setup.bat";
+          (Get-Content $filePath).Replace("/register", "/register /unattended") | Set-Content $filePath
+    - task: CmdLine@2
+      displayName: '[UE] Run Setup.bat'
+      inputs:
+        script: 'ue\Setup.bat --force'
+        workingDirectory: '$(Agent.BuildDirectory)'
+    - task: MSBuild@1
+      displayName: 'Build Plugin'
+      timeoutInMinutes: 300
+      inputs:
+        solution: '$(Build.SourcesDirectory)/build.proj'
+        msbuildArguments: '-p:UnrealEngine=$(Agent.BuildDirectory)\ue;OutputPath=$(Build.ArtifactStagingDirectory)\drop'
+        createLogFile: true
+    - task: PoliCheck@2
+      inputs:
+        targetType: 'F'
+        targetArgument: '$(Build.SourcesDirectory)'
+    - task: ComponentGovernanceComponentDetection@0
+      displayName: 'Component Detection'
+    - task: AntiMalware@4
+      inputs:
+        InputType: 'Basic'
+        ScanType: 'CustomScan'
+        FileDirPath: '$(Build.StagingDirectory)'
+        TreatSignatureUpdateFailureAs: 'Warning'
+        SignatureFreshness: 'UpToDate'
+        TreatStaleSignatureAs: 'Error'
+    - task: CredScan@3
+      inputs:
+        scanFolder: '$(Build.SourcesDirectory)'
+    - task: BinSkim@4
+      inputs:
+        InputType: 'Basic'
+        Function: 'analyze'
+        TargetPattern: 'guardianGlob'
+        AnalyzeTargetGlob: '$(Build.ArtifactStagingDirectory)/drop/**.dll'
+    - task: SDLNativeRules@3

```

Comment:
This does not really do what we need yet, because it's meant for true MSBuild-based flow, and the `build.proj` file in the repo is just a wrapper to call the UE tools. We will have to follow up with the Compliance folks to discuss how to use these with a custom build system.

Reply:
Can some necessary options be force-injected with `_CL_` environment variable?

Reply:
I'm honestly not sure. IIUC what the SDL task is doing, it's injecting MSBuild `.targets` that will configure the CL.EXE command line, so it's really expecting a MSBuild-based build system. We'd need to replicate what those targets are doing, which becomes a maintenance problem.